docs/40_smart-contract-guides/80_linking-custom-permission.md

```diff
@@ -2,72 +2,585 @@
+* A custom permission, _customp1_,  whose parent is active
+* A custom permission, _customp2_, whose parent is _customp1_
+
+You will link the custom permissions to smart contract actions:
+* Link _customp1_ to `what(eosio::name user)`
+* Link _customp2_ to `how(eosio::name user)`
+
+You will see that the `active` permission can call all the actions.
+You will see that the _customp1_  permission can call the `why` and `how` actions.
+You will see that the _customp2_  permission can only call the `how` action.
+
+### The simple smart contract
+
+The simple smart contract code used in the examples:
+
+```c++
+
+#include <eosio/eosio.hpp>
+class [[eosio::contract]] hello : public eosio::contract {
+  public:
+      using eosio::contract::contract;
+      [[eosio::action]] void what( eosio::name user ) {
+         print( "hi, what do you want ", user);
+      }
+
+      [[eosio::action]] void why( eosio::name user ) {
+         print( "why not ", user);
+      }
+
+      [[eosio::action]] void how( eosio::name user ) {
+         print( "how are you ", user);
+      }
+};
+
+```
+
+Deploy this smart contract to your running blockchain.
+
+[[info]]
+| This simple sample  smart contract does not do any authorization checking, i.e. does not use  `require_auth`. This tutorial demonstrates **native authorization checking**. Click on this link for more information on [securing your contract.](https://developers.eos.io/manuals/eosio.cdt/latest/best-practices/securing_your_contract/#1-authorization-checks)
+
+## Create Custom Permissions
+To use a custom permission you need to create a custom permission. 
+
+Create custom permission _customp1_, with the parent `active`, on the bob account using the command [cleos set account permission](https://developers.eos.io/manuals/eos/latest/cleos/command-reference/set/set-account-permission.md):
+
+
+```shell
+cleos set account permission bob customp1 EOS58wmANoBtT7RdPgMRCGDb37tcCQswfwVpj6NzC55D247tTMU9D active -p bob@active
+```
+
+The output should be similar to:
+
+```console
+executed transaction: 97e2af6966b40ea0b523402110c6a5592862c5ad2abbaad20c9bbf2f68017c98  160 bytes  145 us
+#         eosio <= eosio::updateauth            {"account":"bob","permission":"customp1","parent":"active","auth":{"threshold":1,"keys":[{"key":"EOS...
+```
+
+Now create custom permission _customp2_, with the parent _customp1_, on the bob account:
+
+```shell
+cleos set account permission bob customp2 EOS58wmANoBtT7RdPgMRCGDb37tcCQswfwVpj6NzC55D247tTMU9D customp1 -p bob@active
+
+```
+
+The output should be similar to:
+
+```console
+executed transaction: 8b5e88d0d1cea6dd31a4967912d575d62391348345c58b6071aba7fb93d709b3  160 bytes  129 us
+#         eosio <= eosio::updateauth            {"account":"bob","permission":"customp2","parent":"customp1","auth":{"threshold":1,"keys":[{"key":"E...
+```
+
+You can create a custom permission without specifying the parent, this will default to a parent of active:
```

Comment:
For consistency with other instances:
active -> \`active\`

Reply:
updated